contributor-guide/modules/ROOT/nav.adoc

```diff
@@ -33,6 +33,14 @@
 ** xref:testing/continuous-integration.adoc[]
 ** xref:testing/fuzzing.adoc[]
 
+* Modular Boost
```

Comment:
This content was originally written a long time ago and I think this concept of "Modular Boost" has changed over time. At the time, we had SVN boost in a single repository, and then boost git repositories came up. Because each library had its own repository, this was considered modular. Note that this is part of the contributor guide and didn't affect who the users consumed boost.

Nowadays, when people talk about modularizing boost, they are talking about the ability of the _user_ to fetch only the boost modules they need as individual libraries in one way or another. That's why the new user guide includes content about how to download only individual modules with their dependencies and how to consume individual libraries as packages.

This distinction would make things a little confusing for people reading it nowadays. What we used to call "modular" boost at that time, is not enough for people to consider boost modular. The old term is better represented by the "Boost repositories layout" or something like that. 

We should get feedback from @akrzemi1 in this PR.

The commit message should also include `fix #149` in its body.

Reply:
@alandefreitas I have renamed the section "The Git Super-project" - which seems to be what it's about - and have reworked or removed all references to "modular boost" so we can re-use this term when needed.

Reply:
Oh! That's a great name. The super-project is what people are usually trying to understand. I think I would just call it "The Super-Project" because it's more powerful and less is more.

Reply:
@alandefreitas OK, sounds fine, made that change and corrected all titles so that they are title-case, and squashed.